analyses/cell-type-ewings/scripts/utils/tumor-validation-helpers.R

```diff
@@ -0,0 +1,200 @@
+# These are functions used to prepare data frames for plotting when doing tumor cell validataion 
+
+# Data frame prep --------------------------------------------------------------
+
+create_classification_df <- function(
+    sce,
+    marker_gene_results_file, # path to marker gene classifications
+    copykat_predictions_file, # path to copykat predictions
+    infercnv_predictions_file, # path to infercnv predictions
+    infercnv_metadata_file, # path to full infercnv metadata
+    geneset_scores_file # path to gene set scores results
+){
+ 
+  # read in marker gene results 
+  marker_gene_results_df <- readr::read_tsv(marker_gene_results_file)
+  
+  # copykat output
+  copykat_results_df <- readr::read_tsv(copykat_predictions_file) |> 
+    # only select no_ref classification and rename to be consistent with naming of other classifications
+    dplyr::select(barcodes, copykat_classification = no_ref, mean_cnv_detection) |> 
+    dplyr::distinct() |> 
+    dplyr::mutate(
+      copykat_classification = dplyr::case_when(
+        copykat_classification == "aneuploid" ~ "Tumor",
+        copykat_classification == "diploid" ~ "Normal",
+        TRUE ~ copykat_classification
+      )
+    )
+  
+  # infercnv output 
+  infercnv_results_df <- readr::read_tsv(infercnv_predictions_file)
+  infercnv_metadata_df <- read.delim(infercnv_metadata_file) |> 
+    tibble::rownames_to_column("barcodes")
+  
+  # remove MT and GL chr, comments in docs recommend that these are not useful and can be inaccurate 
+  chr_columns_to_keep <- which(!stringr::str_detect(names(infercnv_metadata_df), "chrMT|chrGL"))
+  infercnv_metadata_df <- infercnv_metadata_df[, chr_columns_to_keep]
+  
+  # gene set scores 
+  geneset_scores_df <- readr::read_tsv(geneset_scores_file)
+  
+  # pull out the UMAP coordinates and make a data frame to use for plotting
+  classification_df <- sce |> 
+    scuttle::makePerCellDF(use.dimred = "UMAP") |> 
+    # replace UMAP.1 with UMAP1
+    dplyr::rename_with(
+      \(x) stringr::str_replace(x, "^UMAP\\.", "UMAP")
+    ) |>
+    # get rid of excess columns
+    dplyr::select(barcodes, UMAP1, UMAP2) |> 
+    # add in marker gene classifications
+    dplyr::left_join(marker_gene_results_df) |> 
+    # add in copykat results 
+    dplyr::left_join(copykat_results_df) |> 
+    # add in infercnv results and metadata
+    dplyr::left_join(infercnv_results_df) |> 
+    dplyr::left_join(infercnv_metadata_df) |> 
+    # gene set scores 
+    dplyr::left_join(geneset_scores_df)
```

Comment:
add in `by = ` statements?
```suggestion
    dplyr::left_join(marker_gene_results_df) |> 
    # add in copykat results 
    dplyr::left_join(copykat_results_df) |> 
    # add in infercnv results and metadata
    dplyr::left_join(infercnv_results_df) |> 
    dplyr::left_join(infercnv_metadata_df) |> 
    # gene set scores 
    dplyr::left_join(geneset_scores_df)
```